toy-submission/main.py

```diff
@@ -39,29 +36,27 @@
 # Configure the logging module
 logging.basicConfig(level=logging.INFO)
 
-quantize = "llm.int8"
+quantize = "bnb.nf4-dq"  # 4-bit NormalFloat with Double-Quantization (see QLoRA paper)
+checkpoint_dir = Path("checkpoints/openlm-research/open_llama_7b")
+precision = "bf16-true"  # weights and data in bfloat16 precision
 
-checkpoint_path = Path("checkpoints/lit-llama/7B/lit-llama.pth")
-tokenizer_path: Path = Path("checkpoints/lit-llama/tokenizer.model")
+fabric = L.Fabric(devices=1, accelerator="cuda", precision=precision)
```

Comment:
The rules only allow using 1 4090 or 1 A100 chips so we can hardcode these values